bdk_electrum_example/src/main.rs

```diff
@@ -89,137 +122,141 @@ fn main() -> anyhow::Result<()> {
-            (new_sparsechain, last_active_indices)
+            )?
         }
         ElectrumCommands::Sync {
-            mut unused,
-            mut unspent,
-            all,
+            mut unused_spks,
+            mut unspent_spks,
+            all_spks,
+            unspent_outpoints,
+            unconfirmed_txs,
             scan_option,
         } => {
-            let (spks, local_chain) = {
-                // Get a short lock on the tracker to get the spks we're interested in
-                let tracker = &*tracker.lock().unwrap();
-                let txout_index = &tracker.txout_index;
-                if !(all || unused || unspent) {
-                    unused = true;
-                    unspent = true;
-                } else if all {
-                    unused = false;
-                    unspent = false
-                }
-                let mut spks: Box<dyn Iterator<Item = bdk_chain::bitcoin::Script>> =
-                    Box::new(core::iter::empty());
-
-                if all {
-                    let all_spks = txout_index
-                        .all_spks()
-                        .iter()
-                        .map(|(k, v)| (k.clone(), v.clone()))
-                        .collect::<Vec<_>>();
-                    spks = Box::new(spks.chain(all_spks.into_iter().map(|(index, script)| {
-                        eprintln!("scanning {:?}", index);
-                        script
-                    })));
-                }
-
-                if unused {
-                    let unused_spks = txout_index
-                        .unused_spks(..)
-                        .map(|(k, v)| (k.clone(), v.clone()))
-                        .collect::<Vec<_>>();
-                    spks = Box::new(spks.chain(unused_spks.into_iter().map(|(index, script)| {
-                        eprintln!("Checking if address at {:?} has been used", index);
-                        script
-                    })));
-                }
-
-                if unspent {
-                    let unspent_txouts = tracker
-                        .full_utxos()
-                        .map(|(k, v)| (k.clone(), v.clone()))
-                        .collect::<Vec<_>>();
-                    spks = Box::new(spks.chain(unspent_txouts.into_iter().map(
-                        |(_index, ftxout)| {
-                            eprintln!("checking if {} has been spent", ftxout.outpoint);
-                            ftxout.txout.script_pubkey
-                        },
-                    )));
-                }
-                let local_chain = tracker.chain().checkpoints().clone();
+            // Get a short lock on the tracker to get the spks we're interested in
+            let tracker = tracker.lock().unwrap();
 
-                (spks, local_chain)
-            };
+            if !(all_spks || unused_spks || unspent_spks || unspent_outpoints || unconfirmed_txs) {
+                unused_spks = true;
+                unspent_spks = true;
```

Comment:
I wonder if there's a better way to enforce that at least one of the args is true, this looks a bit weird

Reply:
Let's defer this!